catalogs/catalog-messaging-kafka/src/main/java/com/datastrato/gravitino/catalog/kafka/KafkaCatalogOperations.java

```diff
@@ -92,39 +123,193 @@ public void initialize(Map<String, String> config, CatalogInfo info) throws Runt
+      throw new RuntimeException("Failed to create topic in Kafka" + ident, e);
+    }
+
+    return KafkaTopic.builder()
+        .withName(ident.name())
+        .withComment(comment)
+        .withProperties(properties)
+        .withAuditInfo(
+            AuditInfo.builder()
+                .withCreator(PrincipalUtils.getCurrentPrincipal().getName())
+                .withCreateTime(Instant.now())
+                .build())
+        .build();
   }
 
   @Override
   public Topic alterTopic(NameIdentifier ident, TopicChange... changes)
       throws NoSuchTopicException, IllegalArgumentException {
-    throw new UnsupportedOperationException();
+    KafkaTopic topic = (KafkaTopic) loadTopic(ident);
+    String newComment = topic.comment();
+    int oldPartitionCount = Integer.parseInt(topic.properties().get(PARTITION_COUNT));
+    int newPartitionCount = oldPartitionCount;
+    Map<String, String> alteredProperties = Maps.newHashMap(topic.properties());
+    List<AlterConfigOp> alterConfigOps = Lists.newArrayList();
+    for (TopicChange change : changes) {
+      if (change instanceof TopicChange.UpdateTopicComment) {
+        newComment = ((TopicChange.UpdateTopicComment) change).getNewComment();
+
+      } else if (change instanceof TopicChange.SetProperty) {
+        TopicChange.SetProperty setProperty = (TopicChange.SetProperty) change;
+        if (PARTITION_COUNT.equals(setProperty.getProperty())) {
+          // alter partition count
+          newPartitionCount = setPartitionCount(setProperty, newPartitionCount, alteredProperties);
+        } else {
+          // alter other properties
+          setProperty(setProperty, alteredProperties, alterConfigOps);
+        }
+
+      } else if (change instanceof TopicChange.RemoveProperty) {
+        removeProperty((TopicChange.RemoveProperty) change, alteredProperties, alterConfigOps);
+
+      } else {
+        throw new IllegalArgumentException("Unsupported topic change: " + change);
+      }
+    }
+
+    if (newPartitionCount != oldPartitionCount) {
+      doPartitionCountIncrement(ident.name(), newPartitionCount);
+    }
+
+    if (!alterConfigOps.isEmpty()) {
+      doAlterTopicConfig(ident.name(), alterConfigOps);
+    }
+
+    return KafkaTopic.builder()
+        .withName(ident.name())
+        .withComment(newComment)
+        .withProperties(alteredProperties)
+        .withAuditInfo(
+            AuditInfo.builder()
+                .withCreator(topic.auditInfo().creator())
+                .withCreateTime(topic.auditInfo().createTime())
+                .withLastModifier(PrincipalUtils.getCurrentPrincipal().getName())
+                .withLastModifiedTime(Instant.now())
+                .build())
+        .build();
   }
 
   @Override
   public boolean dropTopic(NameIdentifier ident) throws NoSuchTopicException {
-    throw new UnsupportedOperationException();
+    try {
+      adminClient.deleteTopics(Collections.singleton(ident.name())).all().get();
```

Comment:
So why don't you check the existence of `schema` as other does?

Reply:
fixed